pkg/skaffold/build/local/prune.go

```diff
@@ -158,6 +158,10 @@ func (p *pruner) collectImagesToPrune(ctx context.Context, artifacts []*latest.A
 
 		imgs, err := p.listImages(ctx, a.ImageName)
 		if err != nil {
+			switch err {
```

Comment:
nice catch!

Reply:
the problem is that all docker client calls will fail immediately with a cancelled context so even if I don't skip here for `DeadlineExceeded` cleanup will still fail with these messages. We would need another context for the pruner and not the common context. I'll add a bug for this.